### ISSUE TITLE
Consider reverting change to revision attribute in pom

### ISSUE THREAD

**Issue by snicoll** [spring-projects/spring-retry#480]
Please consider reverting https://github.com/spring-projects/spring-retry/issues/474 and everything that it caused. The purpose of the GitHub workflow is to make sure that things as much aligned as possible and each deviation from that makes Spring Retry special for no good reason.

The maven flatten plugin can easily be introduced to publish a POM that's more clean and without unnecessary information. That would be a plus as well.

**Comment by artembilan** [comment]
Hey, Stephane!

Thank you for feedback!

So, my logic was like this:
If community complains about `revision` property, then indeed something is wrong.
Then it looks like this `revision` was introduced only for easier way to extract the project version during GHA release WF run.
Since there is a standard Maven way to get that info via `mvn help:evaluate -Dexpression="project.version" -q -DforceStdout` command, there is no reason in extra convenient property.

I'm not familiar with Maven flatten plugin, but introducing it just for `revision` property mitigation feels like an overhead in such a single module project.

And if you talk about alignment and deviations, there is [the reusable workflows project](https://github.com/spring-io/spring-github-workflows) which we use successfully in many places.
It might not be perfect, but it does exactly the same for many CI/CD processes and more.
I will be more than happy to migrate Spring Retry there as well. So, we would have one more project aligned and pointed only to a single place of truth or failure.

We may discuss this further offline.

**Comment by snicoll** [comment]
>  If community complains about revision property, then indeed something is wrong.

Yeah, I am not denying that. I missed to add the last step in https://maven.apache.org/maven-ci-friendly.html, sorry about that.

> Since there is a standard Maven way to get that info via mvn help:evaluate -Dexpression="project.version" -q -DforceStdout command, there is no reason in extra convenient property.

I disagree with the analysis. The revision property has been introduced to mimic many other projects that use it. The link that I've shared above explains the benefit of using such property. The property hasn't been only introduced as a way to to get the version when releasing.

> And if you talk about alignment and deviations, there is [the reusable workflows project](https://github.com/spring-io/spring-github-workflows) which we use successfully in many places.

Sure, it's another reusable workflow you could use rather than this one, I don't have an opinion about that. But that's two different topic. Right now, it's neither.

**Comment by artembilan** [comment]
Thank you for the link, Stephane!
Now I know more about Maven 😄 

So, I'll revert to `revision` and will look into this:

> If you like to install or deploy artifacts by using the above setup you have to use the [flatten-maven-plugin](https://www.mojohaus.org/flatten-maven-plugin/)
